### PR TITLE
update ORCA easyblock for version 6

### PR DESCRIPTION
**Pull request by smoors**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by branfosj** [comment]
Test report by @branfosj
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _ORCA-5.0.4-gompi-2023a.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0105u03a - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/09923d475118f0cbe6cc702903c6f098 for a full test report.

**Comment by sdx23** [comment]
Can confirm successful build using
```
  - ORCA-6.0.0-gompi-2023b.eb:
      options:
        include-easyblocks-from-pr: 3395
```
with the EC being just the updated version of the 5.0.4 EC.

File: easybuild/easyblocks/o/orca.py [hunk 1]
```diff
@@ -86,14 +86,21 @@ def install_step(self):
...
-            # Shared builds have additional libraries
-            libs_to_copy = (['liborca*'], 'lib')
-            if all([glob.glob(p) for p in libs_to_copy[0]]):
-                files_to_copy.append(libs_to_copy)
+
+            # Version 6 extra files
+            if LooseVersion(self.version) >= LooseVersion('6.0.0'):
+                files_to_copy.extend(['datasets', 'lib', (['CompoundScripts'], 'bin')])
+
+            else:
```
Comment (akesandgren) [inline easybuild/easyblocks/o/orca.py:94]: Should this really be an "else"? Are the extra files from version 5 no longer available in version 6?

Comment (smoors) [reply easybuild/easyblocks/o/orca.py:94]: yes it should.
`ORCACompoundMethods` no longer exists, it is now `CompoundScripts`, and `liborca*` is now in a `lib` dir, so we can just copy over `lib`.

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by akesandgren** [comment]
Going in, thanks @smoors!

**Comment by akesandgren** [comment]
Going in, thanks @smoors!

**Comment by rlpitts** [comment]
EB said installation was completed but the user got this error:
/sw/easybuild_milan/software/ORCA/6.0.0-gompi-2023a/bin/orca: error while loading shared libraries: liborca_tools_6_0_0.so.6: cannot open shared object file: No such file or directory
Did I forget to set something in the installation command?

**Pull request merged**